decayAmplitude/massDependence.cc

```diff
@@ -564,6 +617,231 @@ piPiSWaveAuMorganPenningtonKachaev::piPiSWaveAuMorganPenningtonKachaev()
+		if (not pdt.isInTable(partList[i])) {
+			printErr<< "cannot find particle " << partList[i] << " in particle data table. "
+			<< "Aborting..." << endl;
+			throw;
+		}
+	_piChargedMass = pdt.entry("pi+")->mass();
+	_kaonChargedMass = pdt.entry("K+")->mass();
+}
+
+
+
+boost::shared_ptr<KPiSGLASS>
+KPiSGLASS::Create(const libconfig::Setting* massDepKey){
+	if (massDepKey != nullptr) {
+		double a, r, M0, G0, phiF, phiR, phiRsin, F, R, MMax;
+		string tag;
+		if(not massDepKey->lookupValue("a", a)){
+			printErr << "cannot find 'a' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("r", r)){
+			printErr << "cannot find 'r' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("M0", M0)){
+			printErr << "cannot find 'M0' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("G0", G0)){
+			printErr << "cannot find 'G0' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("phiF", phiF)){
+			printErr << "cannot find 'phiF' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("phiR", phiR)){
+			printErr << "cannot find 'phiR' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("phiRsin", phiRsin)){
+			printErr << "cannot find 'phiRsin' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("F", F)){
+			printErr << "cannot find 'F' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("R", R)){
+			printErr << "cannot find 'R' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("MMax", MMax)){
+			printErr << "cannot find 'MMax' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("tag", tag)){
+			printErr << "cannot find 'tag' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		return KPiSGLASS::Create(a, r, M0, G0, phiF, phiR, phiRsin, F, R, MMax, tag);
+	} else {
+		printWarn << "no configuration given for mass dependence '" << Name() << "'. Using dummy config" << endl;
+		return KPiSGLASS::Create(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, "");
+	}
+}
+
+
+std::string
+KPiSGLASS::parentLabelForWaveName(const isobarDecayVertex& v) const{
+	std::ostringstream label;
+	label << name() << "<" << _tag << ">[" << v.parent()->name() << "]";
```

Comment:
Should we include the parameter values?

Reply:
I do not think so, because this would lead to extremely long file names.